lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -823,20 +823,149 @@ void TypeLoweringVisitor::visitExpr(SubfieldOp op) {
   opsToRemove.push_back(op);
 }
 
-// Gracefully die on subaccess operations
 void TypeLoweringVisitor::visitExpr(SubaccessOp op) {
-  op.emitError("SubaccessOp not handled.");
+  Value inputVector = op.input();
+  FIRRTLType vectorElementType = inputVector.getType()
+                                     .cast<FIRRTLType>()
+                                     .cast<FVectorType>()
+                                     .getElementType();
+  // If this is a multi-dim vector, then substitution will happen at the leaf
+  // node, when the non-vector subelement is accessed.
+  if (vectorElementType.isa<FVectorType>()) {
+    opsToRemove.push_back(op);
+    return;
+  }
```

Comment:
What if a multi-dimensional array is subaccessed and operated on in both lower-D space and 1-D space?
```
A[n][m];
connect A[1] = stuff;
connect A[2][3] = otherstuff;
```